ghastcoiler/minions/rank_2.py

```diff
@@ -0,0 +1,201 @@
+from minions.types import MinionType
+
+from deathrattles.rank_2 import HarvestGolemDeathrattle, ImprisonerDeathrattle, KaboomBotDeathrattle, \
+    KindlyGrandmotherDeathrattle, RatPackDeathrattle, SpawnofNZothDeathrattle, UnstableGhoulDeathrattle
+
+
+class GlyphGuardian(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Glyph Guardian",
+                         rank=2,
+                         base_attack=2,
+                         base_defense=4,
+                         types=[MinionType.Dragon],
+                         **kwargs)
+
+    def on_attack(self):
+        self.attack *= 2
+
+
+class HarvestGolem(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Harvest Golem",
+                         rank=2,
+                         base_attack=2,
+                         base_defense=3,
+                         types=[MinionType.Mech],
+                         base_deathrattle=HarvestGolemDeathrattle(),
+                         **kwargs)
+
+
+class Imprisoner(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Imprisoner",
+                         rank=2,
+                         base_attack=3,
+                         base_defense=3,
+                         types=[MinionType.Demon],
+                         base_taunt=True,
+                         base_deathrattle=ImprisonerDeathrattle(),
+                         **kwargs)
+
+
+class KaboomBot(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Kaboom Bot",
+                         rank=2,
+                         base_attack=2,
+                         base_defense=2,
+                         types=[MinionType.Mech],
+                         base_deathrattle=KaboomBotDeathrattle(),
+                         **kwargs)
+
+
+class KindlyGrandmother(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Kindly Grandmother",
+                         rank=2,
+                         base_attack=1,
+                         base_defense=1,
+                         types=[MinionType.Beast],
+                         base_deathrattle=KindlyGrandmotherDeathrattle(),
+                         **kwargs)
+
+
+class MetaltoothLeaper(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Metaltooth Leaper",
+                         rank=2,
+                         base_attack=3,
+                         base_defense=3,
+                         types=[MinionType.Mech],
+                         # TODO : add battlecry
+                         **kwargs)
+
+
+class MurlocWarleader(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Murloc Warleader",
+                         rank=2,
+                         base_attack=3,
+                         base_defense=3,
+                         types=[MinionType.Murloc],
+                         **kwargs)
+
+    def gives_attack_defense_bonus(self, other_minion):
```

Comment:
This gets passed the other_minion argument to check if it gives a bonus to that minion, this means we need to check whether the minion is of the type Murloc, if so give 2, 0 otherwise 0, 0. And if it is golden it should give 4, 0